library/src/tensile_host.cpp

```diff
@@ -0,0 +1,398 @@
+#ifdef USE_TENSILE_HOST
+
+#include "rocblas.h"
+#include "tensile_host.hpp"
+
+
+#include <Tensile/Tensile.hpp>
+#include <Tensile/Contractions.hpp>
+#include <Tensile/EmbeddedLibrary.hpp>
+#include <Tensile/MasterSolutionLibrary.hpp>
+#include <Tensile/hip/HipSolutionAdapter.hpp>
+#include <Tensile/hip/HipHardware.hpp>
+#include <Tensile/hip/HipUtils.hpp>
+
+#include <Tensile/Utils.hpp>
+#include <Tensile/TensorDescriptor.hpp>
+
+#include <string>
+#include <iostream>
+//#include <filesystem>
+
+#include <glob.h>
+
+template <typename>
+Tensile::DataType tensile_datatype()
+{
+    throw std::runtime_error("undefined datatype");
+    return Tensile::DataType::Float;
+}
+
+template <>
+Tensile::DataType tensile_datatype<rocblas_half>()
+{
+    return Tensile::DataType::Half;
+}
+
+template <>
+Tensile::DataType tensile_datatype<float>()
+{
+    return Tensile::DataType::Float;
+}
+ 
+template <>
+Tensile::DataType tensile_datatype<double>()
+{
+    return Tensile::DataType::Double;
+}
+
+template <>
+Tensile::DataType tensile_datatype<rocblas_float_complex>()
+{
+    return Tensile::DataType::ComplexFloat;
+}
+
+template <>
+Tensile::DataType tensile_datatype<rocblas_double_complex>()
+{
+    return Tensile::DataType::ComplexDouble;
+}
+
+template <typename T>
+Tensile::ContractionProblem create_gemm_contraction_problem_strided (
+                                     rocblas_operation trans_a,
+                                     rocblas_operation trans_b,
+                                     unsigned long       m,
+                                     unsigned long       n,
+                                     unsigned long       k,
+                                     const T*          alpha,
+                                     const T*          A,
+                                     unsigned long       ld_a,
+                                     unsigned long     stride_a,
+                                     const T*          B,
+                                     unsigned long       ld_b,
+                                     unsigned long     stride_b,
+                                     const T*          beta,
+                                     T*                C,
+                                     unsigned long       ld_c,
+                                     unsigned long     stride_c,
+                                     unsigned long     batchSize)
+{
+
+    bool transposeA = false;
+    if (trans_a == rocblas_operation_conjugate_transpose)
```

Comment:
How is normal transpose handled?

Conjugate transpose needs to be handled with a [TensorOp](https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Source/lib/include/Tensile/TensorOps.hpp).

Reply:
> 
> 
> How is normal transpose handled?
> 
> Conjugate transpose needs to be handled with a [TensorOp](https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Source/lib/include/Tensile/TensorOps.hpp).

Being refactored, but right now transpose and conjugate transpose are treated the same, and no `TensorOp` is created.